// _ng-dynamic@3.0.2@ng-dynamic/src/dynamic-html/dynamic-html.component.ngfactory.js
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
/* tslint:disable */
import * as import0 from '@angular/core';
import * as import1 from './dynamic-html.component';
import * as import2 from './renderer';
var styles_DynamicHTMLComponent = [];
export var RenderType_DynamicHTMLComponent = import0.ɵcrt({
    encapsulation: 2,
    styles: styles_DynamicHTMLComponent,
    data: {}
});
export function View_DynamicHTMLComponent_0(l) {
    return import0.ɵvid(0, [], null, null);
}
function View_DynamicHTMLComponent_Host_0(l) {
    return import0.ɵvid(0, [
        (l()(), import0.ɵeld(0, null, null, 1, 'dynamic-html', [], null, null, null, View_DynamicHTMLComponent_0, RenderType_DynamicHTMLComponent)),
        import0.ɵdid(483328, null, 0, import1.DynamicHTMLComponent, [
            import2.DynamicHTMLRenderer,
            import0.ElementRef
        ], null, null)
    ], function (ck, v) {
        ck(v, 1, 0);
    }, null);
}
export var DynamicHTMLComponentNgFactory = import0.ɵccf('dynamic-html', import1.DynamicHTMLComponent, View_DynamicHTMLComponent_Host_0, { content: 'content' }, {}, []);
